backend/inspirehep/disambiguation/cli.py

```diff
@@ -76,36 +72,59 @@ def query_authors_with_linked_papers_by_bai(authors_bais):
         yield data.author_id
 
 
+def _get_all_not_disambiguated_records_search():
+    query = {
+        "query": {
+            "bool": {
+                "must": [
+                    {
+                        "nested": {
+                            "path": "authors",
+                            "query": {
+                                "bool": {
+                                    "must_not": {
+                                        "exists": {"field": "authors.record.$ref"}
+                                    }
+                                }
+                            },
+                        }
+                    },
+                    {"match": {"_collections": "Literature"}},
+                ]
+            }
+        }
+    }
+
+    search_obj = (
+        LiteratureSearch()
+        .from_dict(query)
+        .params(track_total_hits=True, _source={}, size=1000, scroll="60m")
+    )
+    return search_obj
+
+
 @disambiguation.command(name="not-disambiguated")
 @with_appcontext
 @click.option(
     "-bs", "--batch-size", type=int, default=5, help="Batch size for celery task chunks"
 )
 def disambiguate_all_not_disambiguated(batch_size):
     """Trigger disambiguation task for all the records that are not disambiguated"""
-    not_disambiguiated_records_condition = (
-        not_(
-            type_coerce(RecordMetadata.json, JSONB)["authors"].has_key("record")  # noqa
-        )
-    ) & (
-        type_coerce(RecordMetadata.json, JSONB)["_collections"].contains(["Literature"])
-    )
-    query = RecordMetadata.query.with_entities(cast(RecordMetadata.id, String)).filter(
-        not_disambiguiated_records_condition
-    )
-    with click.progressbar(
-        query.yield_per(1000),
-        length=query.count(),
-        label="Scheduling disambiguation tasks",
-    ) as items:
-        batch = next_batch(items, 1000)
-        while batch:
-            uuids = [str(item[0]) for item in batch]
-            task_group = disambiguate_authors.chunks(
-                zip(uuids, [True for _ in uuids]), batch_size
-            ).group()
+    not_disambiguated_records_search = _get_all_not_disambiguated_records_search()
+    scan_obj = not_disambiguated_records_search.scan(size=1000)
```

Comment:
This will fail actually, you have to use the helper :P

```
In [10]:     scan_obj = not_disambiguated_records_search.scan(size=2)
TypeError: scan() got an unexpected keyword argument 'size'
```

Shouldn't we catch this in tests or are not run yet?